### DISCUSSION TITLE
Define FK relationships in T4 .tt file?

### DISCUSSION THREAD

**Discussion by Pretagonist** [linq2db/linq2db#3369] [Q&A] [answered]
I'm working against a legacy database that I'm not really interested in modifying. There are a couple of relationships that aren't defined in the DB scheme but I want the T4 system to generate them anyway. Is there any way to do this?

**Comment by MaceWindu** [comment, marked as answer]
Before `GenerateModel()` call, which will generate code from loaded model you can modify it by adding association to table.
Check https://linq2db.github.io/articles/T4.html#useful-members-and-data-structures `Table.ForeignKeys` dictionary and `ForeignKey` class

**Reply by MaceWindu** [reply]
with linq2db.cli it is possible to modify schema information in schema load interceptors https://github.com/linq2db/linq2db/tree/master/NuGet/CLI#schema-load-interceptors
